Dockerfile

```diff
@@ -58,19 +58,28 @@ COPY $SCRIPTPATH/*.repos ./
 RUN chmod +x init/initial_ros_setup.sh
 RUN yes | ./init/initial_ros_setup.sh --no-ros2 --download-only
 
+WORKDIR /ros2_ws/navigation2_ws/src/navigation2
+
 # change to correct branch if $BRANCH is not = master
 ARG BRANCH=master
+ARG REMOTE=origin
+RUN echo "Remote is $REMOTE"
 RUN if [ "$BRANCH" == "master" ]; \
     then \
       echo "On master branch"; \
     else \
-      cd navigation2; \
-      git fetch origin $BRANCH:temp_branch; \
-      git checkout temp_branch; \
-      cd -; \
+      if [ "$REMOTE" == "origin" ]; \
```

Comment:
It might make sense to push all this code to the initial_ros_setup script. It is generally useful and it might be easier to maintain in a shell script rather than a docker file.

Reply:
Yes, I thought about doing that but I didn't want to make the initial_ros_setup script more complicated than it needed to be for now. If this feature becomes useful in the script we can move it later. (famous last words, now watch we'll figure out a reason it needs to be there tomorrow, haha)